### PR TITLE
test(godocfx): add golden test

### PR DESCRIPTION
**Pull request by tbpg**
I did some minor refactoring of the write logic to take it out of `func main` and make it testable.

### CODE REVIEW

File: internal/godocfx/main.go [hunk 1]
```diff
@@ -76,8 +77,13 @@ func main() {
 	if *rm {
 		os.RemoveAll(*outDir)
 	}
-	if err := os.MkdirAll(*outDir, os.ModePerm); err != nil {
-		log.Fatalf("os.MkdirAll: %v", err)
+
+	write(*outDir, pages, toc, module)
```
Comment (codyoss) [inline internal/godocfx/main.go:81]: Error should be logged.

**Review by codyoss — APPROVED** [review]
One small thing, other than that LGTM.

File: internal/godocfx/main.go (hunk 1 above)
Comment (tbpg) [reply internal/godocfx/main.go]: Done.

**Comment by tbpg** [comment]
Thanks!

**Pull request merged**